tests/e2e/tx/benchmarks_test.go

```diff
@@ -0,0 +1,170 @@
+package tx_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cosmossdk.io/simapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	clienttx "github.com/cosmos/cosmos-sdk/client/tx"
+	addresscodec "github.com/cosmos/cosmos-sdk/codec/address"
+	"github.com/cosmos/cosmos-sdk/testutil/cli"
+	"github.com/cosmos/cosmos-sdk/testutil/network"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authclient "github.com/cosmos/cosmos-sdk/x/auth/client"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"gotest.tools/v3/assert"
+)
+
+type E2EBenchmarkSuite struct {
+	cfg     network.Config
+	network *network.Network
+
+	txHeight    int64
+	queryClient tx.ServiceClient
+	txRes       sdk.TxResponse
+}
+
+func BenchmarkTx(t *testing.B) {
+	s := NewE2EBenchmarkSuite(t)
+	t.Cleanup(s.Close)
+	val := s.network.Validators[0]
+	txBuilder := mkTxBuilder(t, s)
+	// Convert the txBuilder to a tx.Tx.
```

Comment:
Can we add newline breaks to make this easier to grok pls?

Reply:
Done.